### PR TITLE
Fixing notes typecast issue

### PR DESCRIPTION
**Pull request by mayankamencherla**
[notes typecase] Hardcoding notes to be an array

### CODE REVIEW

File: src/Entity.php [hunk 1]
```diff
@@ -67,34 +67,51 @@ protected function request($method, $relativeUrl, $data = null)
...
     {
         $entities = static::getDefinedEntitiesArray();
 
-        if (isset($data['entity']))
+        if (self::isNotes($key) === true)
         {
-            if (in_array($data['entity'], $entities))
+            $entity = $data;
+        }
+        else
```
Comment (shashankkumar) [inline src/Entity.php]: else if

File: src/Entity.php [hunk 2]
```diff
@@ -67,34 +67,51 @@ protected function request($method, $relativeUrl, $data = null)
...
         }
     }
 
-    protected static function buildEntity($data)
+    protected static function buildEntity($key, $data)
     {
         $entities = static::getDefinedEntitiesArray();
 
-        if (isset($data['entity']))
+        if (self::isNotes($key) === true)
```
Comment (shashankkumar) [inline src/Entity.php]: make a generic isArrayableAttribute function.
Shift 'notes' to an array in the class.

File: src/Entity.php [hunk 3]
```diff
@@ -138,7 +155,7 @@ public function fill($data)
                     {
                         if (is_array($v))
                         {
-                            $entity = static::buildEntity($v);
+                            $entity = static::buildEntity($key, $v);
```
Comment (shashankkumar) [inline src/Entity.php]: create a separate function for filling a single key => value pair.

File: src/Entity.php (hunk 2 above)
Comment (mayankamencherla) [inline src/Entity.php]: Added

File: src/Entity.php (hunk 1 above)
Comment (mayankamencherla) [inline src/Entity.php]: Refactored

File: src/Entity.php (hunk 3 above)
Comment (mayankamencherla) [inline src/Entity.php]: Refactored. Added a helper for the associate array case as well.

**Comment by mayankamencherla** [comment]
@shashankkumar @73sl4 Kindly review.

File: src/Entity.php [hunk 4]
```diff
@@ -67,15 +67,21 @@ protected function request($method, $relativeUrl, $data = null)
         }
         else
         {
-            return static::buildEntity($response);
+            return static::buildEntity(null, $response);
```
Comment (ccrims0n) [inline src/Entity.php]: @captn3m0 not sure if it's a good idea to pass null. Was thinking if we can make key as the second argument. What do you think?

**Comment by harman28** [comment]
Will reopen if needed.

**Pull request closed without merging**